app/src/main/java/src/Authorization/data/LoginDataSource.kt

```diff
@@ -0,0 +1,24 @@
+package src.Authorization.data
+
+import src.Authorization.data.model.LoggedInUser
+import java.io.IOException
+
+/**
+ * Class that handles authentication w/ login credentials and retrieves user information.
+ */
+class LoginDataSource {
+
+    fun login(username: String, password: String): Result<LoggedInUser> {
+        try {
+            // TODO: handle loggedInUser authentication
+            val fakeUser = LoggedInUser(java.util.UUID.randomUUID().toString(), "Jane Doe")
+            return Result.Success(fakeUser)
+        } catch (e: Throwable) {
+            return Result.Error(IOException("Error logging in", e))
```

Comment:
is this a test case? more comments are necessary to understand what is going on here